persist/fs/files.go

```diff
@@ -42,11 +42,103 @@ var timeZero time.Time
 
 const (
 	dataDirName       = "data"
+	snapshotDirName   = "snapshots"
 	commitLogsDirName = "commitlogs"
 )
 
 type fileOpener func(filePath string) (*os.File, error)
 
+// FilesetFile represents a set of Fileset files for a given block start
+type FilesetFile struct {
+	BlockStart time.Time
+	Files      []string
+}
+
+// FilesetFilesSlice is a slice of FilesetFile
+type FilesetFilesSlice []FilesetFile
+
+// Flatten flattens a slice of FilesetFiles to a single slice of filepaths
+func (f FilesetFilesSlice) Flatten() []string {
```

Comment:
nit: maybe just `Filepaths()`? it conveys more information about the return type as opposed to the implementation

Reply:
yeah fair I like it

Reply:
done